datatypes/src/collections/feature_collection.rs

```diff
@@ -88,6 +88,36 @@ impl<CollectionType> AsRef<FeatureCollection<CollectionType>>
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct FeatureCollectionInternals<G> {
```

Comment:
why do you not use this in the `FeatureCollection<CollectionType>`? How can we guarantee it's the same otherwise?

Reply:
this construct helps to transform the data object into a data struct and back. Don't know if that is a good pattern but it works.